python/ribasim/ribasim/model.py

```diff
@@ -232,7 +235,72 @@ def reset_contextvar(self) -> "Model":
         return self
 
     def plot_control_listen(self, ax):
-        raise NotImplementedError()
+        df_listen_edge = pd.DataFrame(
+            data={
+                "control_node_id": pd.Series([], dtype="int"),
+                "control_node_type": pd.Series([], dtype="str"),
+                "listen_node_id": pd.Series([], dtype="int"),
+                "listen_node_type": pd.Series([], dtype="str"),
+            }
+        )
+
+        # Listen edges from PidControl
+        for table in [self.pid_control.static.df, self.pid_control.time.df]:
+            if table is None:
+                continue
+
+            to_add = table[
+                ["node_id", "listen_node_id", "listen_node_type"]
+            ].drop_duplicates()
+            to_add.columns = ["control_node_id", "listen_node_id", "listen_node_type"]
+            to_add["control_node_type"] = "PidControl"
+            df_listen_edge = pd.concat([df_listen_edge, to_add])
+
+        # Listen edges from DiscreteControl
+        condition = self.discrete_control.condition.df
+        if condition is not None:
+            to_add = condition[
+                ["node_id", "listen_node_id", "listen_node_type"]
+            ].drop_duplicates()
+            to_add.columns = ["control_node_id", "listen_node_id", "listen_node_type"]
+            to_add["control_node_type"] = "DiscreteControl"
+            df_listen_edge = pd.concat([df_listen_edge, to_add])
+
+        # Collect geometry data
+        node = self.node_table().df
+        control_nodes_geometry = gpd.GeoSeries(
+            df_listen_edge.merge(
+                node,
+                left_on=["control_node_id", "control_node_type"],
+                right_on=["node_id", "node_type"],
+                how="left",
+            )["geometry"]
+        )
+        listen_nodes_geometry = gpd.GeoSeries(
+            df_listen_edge.merge(
+                node,
+                left_on=["listen_node_id", "listen_node_type"],
+                right_on=["node_id", "node_type"],
+                how="left",
+            )["geometry"]
+        )
+
+        # Plot listen edges
+        line_segments = LineCollection(
+            np.hstack(
+                [
+                    listen_nodes_geometry.x.to_numpy().reshape(-1, 1),
+                    listen_nodes_geometry.y.to_numpy().reshape(-1, 1),
+                    control_nodes_geometry.x.to_numpy().reshape(-1, 1),
+                    control_nodes_geometry.y.to_numpy().reshape(-1, 1),
+                ]
+            ).reshape(-1, 2, 2),
+            color="gray",
+            ls="--",
+            label="Listen edge",
+        )
+        ax.add_collection(line_segments)
```

Comment:
Let's make a loop out of that (or watch @Huite optimize it into something that is fast and readable)